sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -400,12 +407,12 @@ def tune(
         trial_template = models.V1beta1TrialTemplate(
             primary_container_name=constants.DEFAULT_PRIMARY_CONTAINER_NAME,
             retain=retain_trials,
-            trial_parameters=trial_params,
+            trial_parameters=trial_params if callable(objective) else [],
```

Comment:
`trial_parameters` still be required even when user sets Docker image.
You can check example here: https://github.com/kubeflow/katib/blob/master/examples/v1beta1/hp-tuning/random.yaml#L31-L36

Reply:
Sure @andreyvelich , I will revert this change and keep the trial_parameters.

Reply:
Also, I would like to know if we need to write new unit test cases or change existing ones ?

Reply:
cc @tariq-hasan